packages/flutter_goldens/lib/skia_client.dart

```diff
@@ -582,19 +582,22 @@ class SkiaGoldClient {
   /// the image keys.
   String getTraceID(String testName) {
     final String? webRenderer = _webRendererValue;
+    // Keys must be sorted alphabetically, after grouping by letter case.
     final Map<String, Object?> keys = <String, Object?>{
+      // Upper case keys
+      'Abi': abi.toString(),
       if (_isBrowserTest)
         'Browser' : _browserKey,
       if (webRenderer != null)
         'WebRenderer' : webRenderer,
       'CI' : 'luci',
       'Platform' : platform.operatingSystem,
-      // 'Abi': abi.toString(), workaround for https://g-issues.skia.org/issues/339508268
-      // Flutter tracking issue: https://github.com/flutter/flutter/issues/148022
-      'name' : testName,
-      'source_type' : 'flutter',
+
+      // Lower case keys
       if (_isImpeller)
         'impeller': 'swiftshader',
+      'name' : testName,
+      'source_type' : 'flutter',
     };
     final String jsonTrace = json.encode(keys);
```

Comment:
Could we add an assert here (or a test elsewhere) that enforces the sorting rules and breaks if this is messed up in the future?

Reply:
I can, it would be pretty complicated though (and probably not today) to account for all the cases where we do or don't include a given key. We don't add keys very often, I was kind of hoping the comment would suffice. WDYT?

Reply:
It does seem a touch fragile. Would it be worth adding some code that somehow alphabetizes the map keys first? If so, then I would imagine it would be easier to add a few tests for that logic: input(map) output(md5 hash).

Reply:
Isn't this just:

```dart
  assert(() {
    List sorted = keys.keys.toList()..sort();
    return listEquals(sorted, keys.keys.toList());
  }());
```

Although, maybe doing this on every golden check is kinda slowing things down? I'd be okay with leaving this out as well.

Reply:
The key part is removablee

Reply:
> Isn't this just:
> 
> ```dart
>   assert(() {
>     List sorted = keys.keys.toList()..sort();
>     return listEquals(sorted, keys.keys.toList());
>   }());
> ```
> 
> Although, maybe doing this on every golden check is kinda slowing things down? I'd be okay with leaving this out as well.

So assert is like......pulling

Reply:
Another option would be to just throw these key-values into a self-ordering data structor, e.g. https://main-api.flutter.dev/flutter/dart-collection/SplayTreeMap-class.html

Reply:
Oh! Will dart sort by case before alphabetical order? I didn't think it would.. let me check